src/common/cockpitfdpassing.c

```diff
@@ -119,6 +121,11 @@ cockpit_socket_receive_fd (int  socket_fd,
                         .msg_controllen = CMSG_LEN (sizeof *out_fd) };
   assert (msg.msg_controllen <= sizeof cmsg);
 
+  int cur_flags = fcntl (socket_fd, F_GETFL);
```

Comment:
So, "no" to this part, at least, on technical grounds: this is a heavy hammer of an operation to work around a very specific issue.  I'd prefer if we solved that issue by using `poll()` here (and `select.select([0], [], [])` in Python) to block until stdin is readable, instead.

Reply:
Changed to poll(). But OOI (and since you didn't reply in chat):

> stdin does not belong to us / "heavy hammer":

What does that mean? I thought that each fd has an independent blocking/nonblocking setting, and that was more or less just a local choice which kind of API a program wants to use. It certainly shouldn't affect the "other end" (i.e. sudo's side of what the bridge's stdin is connected to).

Reply:
It's true that we own our own file descriptor 0, and this is local for our process ,but that's not how this works.  The non-blocking setting is a property of the file description, not the file descriptor.  It is shared by everyone who has (or will ever have in the future) our stdin.

This is the source of all of the problems we used to have with various node/webpack-based processes changing the non-blocking settings on the terminal.  This shouldn't be the way it is, but it is... :(

Reply:
Right, that's what I thought. So while this locally fixes it for cockpit_socket_receive_fd(), it would remain non-blocking for everything else. Originally I wanted to move this into the main() bridge setup. But you said that most of the bridge gets along with non-blocking stdin (as it's part of the async I/O from the main loop, and thus uses select()/epoll()/friends). Programs run from channels should also be isolated, as cockpit_pipe_spawn() creates new stdin fds for the child. So after that discussion, (1) I agree that the local fix suffices, and (2) I think we share an understanding what the blocking flag does, and I wasn't missing some weird magic that would influence other processes.